cli/src/commands/explore/runtime_apis/mod.rs

```diff
@@ -0,0 +1,204 @@
+use scale_typegen_description::type_description;
+use scale_value::Value;
+use subxt::{
+    ext::{scale_decode::DecodeAsType, scale_encode::EncodeAsType},
+    Metadata,
+};
+use subxt_metadata::RuntimeApiMetadata;
+
+/// Runs for a specified runtime API trait.
+/// Cases to consider:
+/// ```norun
+/// method is:
+///   None => Show pallet docs + available methods
+///   Some (invalid) => Show Error + available methods
+///   Some (valid)   => Show method docs + output type description
+///                       exectute is:
+///                         false => Show input type description + Example Value
+///                         true  => validate (trailing args + build node connection)
+///                           validation is:  
+///                             Err => Show Error
+///                             Ok  => Make a runtime api call witht the provided args.
+///                               response is:
+///                                 Err => Show Error
+///                                 Ok  => Show the result
+/// ```
+pub async fn run<'a>(
+    method: Option<String>,
+    execute: bool,
+    trailing_args: Vec<String>,
+    runtime_api_metadata: RuntimeApiMetadata<'a>,
+    metadata: &'a Metadata,
+    file_or_url: FileOrUrl,
+    output: &mut impl std::io::Write,
+) -> color_eyre::Result<()> {
+    let api_name = runtime_api_metadata.name();
+
+    let usage = || {
+        let methods = methods_to_string(&runtime_api_metadata);
+        formatdoc! {"
+        Usage:
+            subxt explore api {api_name} <METHOD>
+                explore a specific runtime api method
+        
+        {methods}
+        "}
+    };
+
+    // If method is None: Show pallet docs + available methods
+    let Some(method_name) = method else {
+        let doc_string = first_paragraph_of_docs(runtime_api_metadata.docs()).indent(4);
+        if !doc_string.is_empty() {
+            writedoc! {output, "
+            Description:
+            {doc_string}
+    
+            "}?;
+        }
+        writeln!(output, "{}", usage())?;
+        return Ok(());
+    };
+
+    // If method is invalid: Show Error + available methods
+    let Some(method) = runtime_api_metadata
+        .methods()
+        .find(|e| e.name().eq_ignore_ascii_case(&method_name))
+    else {
+        return Err(eyre!(
+            "\"{method_name}\" method not found for \"{method_name}\" runtime api!\n\n{}",
+            usage()
+        ));
+    };
+    // redeclare to not use the wrong capitalization of the input from here on:
+    let method_name = method.name();
+
+    // Method is valid. Show method docs + output type description
+    let doc_string = first_paragraph_of_docs(method.docs()).indent(4);
+    if !doc_string.is_empty() {
+        writedoc! {output, "
+        Description:
+        {doc_string}
+        
+        "}?;
+    }
+
+    #[allow(non_snake_case)]
+    let INPUT_VALUE: String = "<INPUT_VALUE>".blue().to_string();
```

Comment:
Personally I'd just name it something like `blue_input_placeholder` or something (likewise elsewhere) and avoid needing to add the allow